keyboards/ploopyco/trackball_mini/trackball_mini.c

```diff
@@ -0,0 +1,246 @@
+    }
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t* record) {
+    xprintf("KL: kc: %u, col: %u, row: %u, pressed: %u\n", keycode, record->event.key.col, record->event.key.row, record->event.pressed);
+
+    // Update Timer to prevent accidental scrolls
+    if ((record->event.key.col == 1) && (record->event.key.row == 0)) {
+        lastMidClick = timer_read();
+        is_scroll_clicked = record->event.pressed;
+    }
+
+    if (!process_record_user(keycode, record))
+        return false;
+
+    if (keycode == DPI_CONFIG && record->event.pressed) {
+        keyboard_config.dpi_config = (keyboard_config.dpi_config + 1) % DPI_OPTION_SIZE;
+        eeconfig_update_kb(keyboard_config.raw);
+        adns_set_cpi(dpi_array[keyboard_config.dpi_config]);
+    }
+
+/* If Mousekeys is disabled, then use handle the mouse button
+ * keycodes.  This makes things simpler, and allows usage of
+ * the keycodes in a consistent manner.  But only do this if
+ * Mousekeys is not enable, so it's not handled twice.
+ */
+#ifndef MOUSEKEY_ENABLE
+    if (IS_MOUSEKEY_BUTTON(keycode)) {
+        report_mouse_t currentReport = pointing_device_get_report();
+        if (record->event.pressed) {
+            currentReport.buttons |= 1 << (keycode - KC_MS_BTN1);
+        } else {
+            currentReport.buttons &= ~(1 << (keycode - KC_MS_BTN1));
+        }
+        pointing_device_set_report(currentReport);
+        pointing_device_send();
+    }
+#endif
+
+    return true;
+}
+
+// Hardware Setup
+void keyboard_pre_init_kb(void) {
+    // debug_enable = true;
+    // debug_matrix = true;
+    debug_mouse = true;
+    // debug_encoder = true;
+
+    setPinInput(OPT_ENC1);
+    setPinInput(OPT_ENC2);
+
+    /* Ground all output pins connected to ground. This provides additional
+     * pathways to ground. If you're messing with this, know this: driving ANY
+     * of these pins high will cause a short. On the MCU. Ka-blooey.
+     */
+#ifdef UNUSED_PINS
+    const pin_t unused_pins[] = UNUSED_PINS;
+
+    for (uint8_t i = 0; i < (sizeof(unused_pins) / sizeof(pin_t)); i++) {
+        setPinOutput(unused_pins[i]);
+        writePinLow(unused_pins[i]);
+    }
+#endif
+
+    keyboard_pre_init_user();
+}
+
+void pointing_device_init(void) {
+    adns_init();
+    opt_encoder_init();
+}
+
+bool has_report_changed (report_mouse_t first, report_mouse_t second) {
+    return !(
+        (!first.buttons && first.buttons == second.buttons) &&
+        (!first.x && first.x == second.x) &&
+        (!first.y && first.y == second.y) &&
+        (!first.h && first.h == second.h) &&
+        (!first.v && first.v == second.v) );
+}
+
+void pointing_device_task(void) {
+    report_mouse_t mouse_report = pointing_device_get_report();
+    process_wheel(&mouse_report);
+    process_mouse(&mouse_report);
+
+    pointing_device_set_report(mouse_report);
+    if (has_report_changed(mouse_report, pointing_device_get_report())) {
+        pointing_device_send();
+    }
+}
```

Comment:
This logic is built into `pointing_device_send()` by default now. Can be slimmed down to: 
```suggestion
void pointing_device_task(void) {
    report_mouse_t mouse_report = pointing_device_get_report();
    process_wheel(&mouse_report);
    process_mouse(&mouse_report);

    pointing_device_set_report(mouse_report);
    pointing_device_send();
}
```